pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +272,42 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns a list of users encoded in SPIFFE format with the local
+// trust domain and its aliases.
+// For example, for a user "bar" in namespace "foo".
+// If the local trust domain is "td2" and its alias is "td1" (migrating from td1 to td2),
+// getPrincipalsIncludingAliases returns ["td2/ns/foo/sa/bar", "td2/ns/foo/sa/bar]].
+func getPrincipalsIncludingAliases(trustDomain string, trustDomainAliases []string, principals []string) []string {
```

Comment:
1) The function name and comments seems confusing, how about rename it to replaceTrustDomainAliases and update the comment to make it clear it checks and replaces the trust domain aliases in the principals

2) Instead of passing the trustDomain and trustDomainAliases seperately, could you create a struct to include both of them and pass the struct instead?

3) It's better to put these in a new file (or at least move it to util.go), you can also define this function as a method of the struct.

Reply:
@pitlv2109 If you are not going to address this comment now, could you add a TODO and make sure to refactor the code in a follow PR?

Reply:
I thought I did? `TODO(pitv2109): Put |trustDomain| and |trustDomainAliases| together. Do this and for other functions.`

Reply:
I didn't see that in the previous commit but now it's appearing in the PR, thanks.